### DISCUSSION TITLE
Few photometric interpretations are not supported

### DISCUSSION THREAD

**Discussion by anish-tricog** [fo-dicom/fo-dicom#1247] [Q&A]
**Describe the bug**
I checked the code and realized that not all photometric interpretations are supported by fo-dicom. I started checking on this when YBR_Partial_420 type was failing with the following error:
`Dicom.Imaging.DicomImagingException: Unsupported pipeline photometric interpretation: YBR Partial 4:2:0`

**To Reproduce**
Use a dicom file with YBR Partial 4:2:0 Photometric interpretation and try to generate the media file from it (image/frame images)

**Expected behavior**
All photometric interpretation should be supported

**Screenshots or test DICOM files**
[YBR_Partial_420 Files.zip](https://github.com/fo-dicom/fo-dicom/files/7182876/YBR_Partial_420.Files.zip)

**Environment**
Fellow Oak DICOM version: 4.0.6
OS: Windows 10 x64
Platform: .NET Framework 4.7.2 (supported by code but system version is 4.8)

**Comment by mrbean-bremen** [comment]
I'm not sure if just handling this transfer syntax as RGB (like YBRPartial422) will work, but even if it does, the example files cannot be processed - we get:
```
Support for multi-frame images with varying fragment sizes and no offset table has not been implemented.
```
Looks like these are independent issues.

**Comment by anish-tricog** [comment]
In case if you guys need more files to check the issue, I can send them.. I have multiple files with this issue. I thought a few will be enough to fix it. Please let me know in case if more files are needed. I will help you with whatever data is possible for me to get in order to resolve this.

**Comment by mrbean-bremen** [comment]
Thank you - are these files all with MPEG transfer syntax? It would be interesting if you had other files (e.g. not MPEG-encoded) with the same problem.

**Comment by anish-tricog** [comment]
I connected with my team and noticed that all of the files related to this issue are of MPEG transfer syntax.
I could not get my hands on other transfer syntax having the similar issue.
Currently, I have all of them having the transfer syntax UID **`1.2.840.10008.1.2.4.102`** with the value of **`MPEG-4 AVC/H.264 High Profile / Level 4.1`**

**Comment by mrbean-bremen** [comment]
Ok, thank you. I guess the first thing to do is to implement support for images with varying fragment sizes and no offset table, but I'm not sure if this will be done anytime soon - I myself have only very limited knowledge here, but I do remember that this is a bit complicated. I think the issue with YBR_Partial_420 is a secondary one, and could be handled after we have some valid example data.

**Comment by mrbean-bremen** [comment]
Also, the MPEG codec is not supported, so this is something that `fo-dicom` probably won't handle by itself, and I see no indication that Efferent Health plans to add such a codec. The mentioned message about the offset table may also be wrong for this case, as MPEG seems to have only one fragment for all frames (by I may be wrong here).
So I would say you are out of luck in this case, as this will not be available in the near future (@gofal - correct me if I'm wrong).

**Comment by mrbean-bremen** [comment]
@anish-tricog - Can you give a bit more background on what are you trying to do? As these are video files, it may not make sense to handle them like other image files, and maybe you can solve your problem in another way.

**Reply by anish-tricog** [reply]
I am trying to read the frames based on the number of frames and then converting the frame images into a video using FFmpeg.
My goal is to have a compressed video extracted from the Pixel Data of the Dicom file. I can do the compression separately if I can either get the frames or the entire video from the pixel data.

**Comment by gofal** [comment]
I think you have no chance to retrieve single frames from this file with fo-dicom. The Mpeg transfersyntax means, that the pixeldata is some encapsulated stream and contains a Mpeg-4 h.264 encoded video file. You could take the whole pixel data stream, write it on a harddisk and then use any videoplayer to open that file.
That means, the frames are not stored separately, but the pixeldata first has the mpeg video header, then you have the video data with only full data every X frames and some b-, p- or i-frames for differences between frames in between etc. 
So opening a video stream and extracting frames out of it is out of scope of fo-dicom.

I used to take aforge.net for such a task some time ago. I extracted the whole pixeldata, and then used the package aforge.video to open this stream and retrieve the single frames.

**Reply by anish-tricog** [reply]
I am in fact trying to get the frames for the video. I didn't pay attention to the fact that MPEG encoding doesn't support proper frame decoding as they are just vectored on top of certain frames.
But if we need to get the entire video from the pixel data stream, is it possible to do so with fo-dicom or will I need to use another library for getting a direct video file from the pixel data?

What I am doing is that I am trying to get the frames for a video and use FFmpeg to stitch them back into a video with some compression parameters. If there is a way to get the video directly, it will be even better. How should I go ahead with this?
Any help is appreciated regarding this.

**Reply by mrbean-bremen** [reply]
As @gofal wrote, this is out of scope for fo-dicom. Apart from the other problems (unhandled Photometric Interpretation and unhandled case for encapsulation) which could be worked around, we just don't support that codec, and you will probably be better off using  another library for that. You could use fo-dicom to retrieve the raw encoded pixel data from the DICOM file, but have to process the data using something else - at least this is my understanding so far.
